File: python/monitor.py
import splunklib.client as client
import splunklib.results as results
import utils, sys, time

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) < 2:
    print "Syntax: monitor.py \"search query\" [rt]"
    sys.exit()
try: 
    service = client.connect(**opts.kwargs)
 
    if len(sys.argv) > 2 and (sys.argv[2] == 'rt'):
        params = {#LAB: add name/value pairs for auto cancel in 5 minutes and real-time search 
        print "realtime"
    else:
        params = {#LAB: add name/value pair for auto cancel in 5 minutes
        print "normal"

    job = #LAB: create the job using the query in sys.argv[1] and params
  
    offset = 0
    while True:
        job.refresh()
        total = int(job['eventCount'])
        if total > offset:
            resultSet = job.events(**{"offset" : offset, "count" : total})
            events = results.ResultsReader(resultSet)
            for event in events:
                print event['_raw']
            offset = total
	      		if job.is_done(): break
        else:
            time.sleep(0.1)
                        
except KeyboardInterrupt:
    print "bye"

except Exception, err:
    sys.stderr.write("Exception: " + str(err))
    
finally:
    job.cancel()

File: java/monitor.java
import com.splunk.*; 
import java.io.*;

public class monitor {

	public static void main (String[] args) {
		if(args.length < 1) {
			System.out.println("syntax: monitor \"search query\" [rt]");
			System.exit(0);
		}	
		Job job = null;
		try {
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);
	        
	        JobArgs ja = new JobArgs();
                ja.put("auto_cancel",10);
	    	//LAB: add name/value pair to ja for auto cancel set to 5 minutes 
	        if(args.length == 2 && args[1].equalsIgnoreCase("rt")) { 
    	        //LAB: add name/value pairs to ja for realtime search
                  ja.put("earliest_time","rt");
                  ja.put("latest_time","rt");
	        }
	        
			job = service.getJobs().create(args[0],ja);//LAB: create a search job using the query in args[0] and ja

			int offset = 0;
			Event event;
			
			while (true) {
				int total = job.getEventCount();//LAB: get the total number of events in job so far
				if (total > offset) {
                                        Args outputArgs = new Args("count", total);
                                        outputArgs.put("offset", offset);
					//LAB: add name/value pairs outputArgs for offset=offset and count=total 
					InputStream resultSet = job.getEvents(outputArgs);//LAB: get the job's events, passing outputArgs
					ResultsReaderXml events = new ResultsReaderXml(resultSet);//LAB: convert resultSet to a ResultsReaderXML

					while ((event = events.getNextEvent()) != null) 
						System.out.println(event.get("_raw"));
					offset = total;
				} else Thread.sleep(100);
				if(job.isDone()) break;
			}
		} catch (Exception e) {
			System.err.println(e.getMessage());
		} finally{
			job.cancel();
			
		}
	}
}
